finetune/lora.py

```diff
@@ -193,6 +194,10 @@ def train(
 
         input_ids, targets = get_batch(fabric, train_data, longest_seq_ix if iter_num == 1 else None)
 
+        # Truncate if needed
+        input_ids = input_ids[:, :model.max_seq_length]
+        targets = targets[:, :model.max_seq_length]
```

Comment:
This could be made part of `get_batch` so that we truncate as soon as possible and reduce how much data is padded, moved to device etc.